### DISCUSSION TITLE
v3.0.1 关于 “添加到Aria2下载”，是我的理解有问题吗？

### DISCUSSION THREAD

**Discussion by wuhgit** [AlistGo/alist#1785] [Q&A] [answered]
v3.0.1， 已配置好 Aria2 ，并且能正确识别。

在网盘目录下随机选择一个文件，右键复制链接，右下角点击添加到 Aria2 下载，粘贴刚刚的链接，然后在后台管理里可以看见下载任务。

但是文件下载成功后这个文件又被上传到了之前的网盘目录里面，下载到本地的文件也不见了……

对于这个功能，我的理解是复制链接添加到Aria2下载之后就会把文件的链接发送到 Aria ，后者将这个文件下载到本地。

并且，因为我已经在 Aria2 的启动配置文件中设置了下载目录（dir），发送链接过去之后应该是下载到我定义好的下载目录中，而不是下载到 alist 的临时文件夹里。

另外一个问题是，文件复制链接得到的并不完整，缺少了网址前缀，比如我的 alist 是 `https://<DOMAIN>/alist/` ，得到的链接地址是 `/alist/d/<DIR>/...<FILENAME>` 

有没有可能做到直接右键发送到 Aria2 中进行下载，而不是要自己再复制粘贴一番。

**Comment by xhofe** [comment, marked as answer]
是的，你说的地方的功能是离线下载。

要下载到本地：
![image](https://user-images.githubusercontent.com/36558727/192144163-4d2ebdc2-704e-43a9-9e27-872bcd08c617.png)

**Reply by wuhgit** [reply]
谢谢！完全没想到是在这里……

**Comment by tw2266** [comment]
请问你这个aria怎么配置的 我的只能用ws。http不会设置。 请教一下！

**Reply by wuhgit** [reply]
用 Nginx 反代 aria2 端口就可以了

**Comment by KITblue** [comment]
每次下载都会多会生成一级目录，是正常的吗？
![无标题](https://user-images.githubusercontent.com/68754977/208232215-23d59b44-b26e-49aa-aca1-e0250895fcbf.png)
